### ISSUE TITLE
Error setting up entry mawaqit for mawaqit_prayers_times

### ISSUE THREAD

**Issue by MehdiTouns31** [mawaqit/home-assistant#10]
Salemoua3laykoum,

I tried to reinstall in HA the last version of mawaqit integration, but the mawaqit configuration fail...
This error is displayed in the log:

This error comes from a custom integration Logger: homeassistant.config_entries Source: custom_components / mawaqit_prayer_times / __ init__.py:286 Integration: Mawaqit First occurred: 21:07:56 (2 occurrences) Last logged: 21:17:36 Error setting up entry Mawaqit for mawaqit_prayer_times Traceback (most recent call last): File "/config/custom_components/mawaqit_prayer_times/__init__.py", line 285, in async_setup await self.hass.async_add_executor_job (self.get_new_prayer_times) File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run result = self.fn (* self.args, ** self.kwargs) File "/config/custom_components/mawaqit_prayer_times/__init__.py", line 115, in get_new_prayer_times calc = MawaqitTimesCalculator ( File "/usr/local/lib/python3.8/site-packages/mawaqit_times_calculator/pray_times_calculator.py", line 31, in __init__ if self._mosquee == '' and len (self.all_mosques_neighberhood ())> 0: File "/usr/local/lib/python3.8/site-packages/mawaqit_times_calculator/pray_times_calculator.py", line 66, in all_mosques_neighberhood api = self.apimawaqit () File "/usr/local/lib/python3.8/site-packages/mawaqit_times_calculator/pray_times_calculator.py", line 54, in apimawaqit return response.json () ["apiAccessToken"] File "/usr/local/lib/python3.8/site-packages/requests/models.py", line 900, in json return complexjson.loads (self.text, ** kwargs) File "/usr/local/lib/python3.8/site-packages/simplejson/__init__.py", line 525, in loads return _default_decoder.decode (s) File "/usr/local/lib/python3.8/site-packages/simplejson/decoder.py", line 370, in decode obj, end = self.raw_decode (s) File "/usr/local/lib/python3.8/site-packages/simplejson/decoder.py", line 400, in raw_decode return self.scan_once (s, idx = _w (s, idx) .end ()) simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0) During handling of the above exception, another exception occurred: Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/config_entries.py", line 269, in async_setup result = await component.async_setup_entry (hass, self) # type: ignore File "/config/custom_components/mawaqit_prayer_times/__init__.py", line 72, in async_setup_entry if not await client.async_setup (): File "/config/custom_components/mawaqit_prayer_times/__init__.py", line 286, in async_setup except (exceptions.InvalidResponseError, ConnError) as err: NameError: name 'exceptions' is not defined
"

Since this problem appeared all the prayer sensors are no longer available... 

Can you take a look ?
Barrakallahufikoum.

![Screenshot_20210522_214108](https://user-images.githubusercontent.com/73989636/119239092-0e2f8700-bb47-11eb-83c3-03041b57a498.jpg)
![Screenshot_20210522_214056](https://user-images.githubusercontent.com/73989636/119239095-18518580-bb47-11eb-844d-e9ad0a052bee.jpg)

**Comment by MehdiTouns31** [comment]
@yac69
My GPS coordinates is defined in my configuration file. These one fill automatically in the form as you can notice in this screenshot : 
![Screenshot_20210525_204508](https://user-images.githubusercontent.com/73989636/119555036-af177f80-bd9d-11eb-8d74-672af2abeb6f.jpg)

I am sure my login and password are right because the mawaqit configuration succeed : 
![Screenshot_20210525_203819_com huawei himovie overseas](https://user-images.githubusercontent.com/73989636/119558105-4df1ab00-bda1-11eb-94eb-479a3c38a955.jpg)

I hope you will help me. 
See you.

**Comment by yac69** [comment]
Salam Mehdi
Can you run the piece of code below on your server hassio in bash (or in any ubuntu server where python3 is installed)
just copy paste it to a file, name it test.py for instance
then run python3 test.py

change your mawaqit login and password before running it;

normally, you should see you mawaqit api and the prayer time in a mosque nearest to the gps coordinates.

let me know if you see any error. 

`import os
os.system('pip3 install mawaqit_times_calculator')
from mawaqit_times_calculator import MawaqitTimesCalculator



mawaqit_login = "login here"
mawaqit_password = "password here"
mawaqit_latitude = "45.764043"
mawaqit_longitude = "4.835659"



 
calc = MawaqitTimesCalculator(
    mawaqit_latitude,
    mawaqit_longitude,
    '',
    mawaqit_login,
    mawaqit_password,
    ''
        )

print(calc.apimawaqit())
print(calc.fetch_prayer_times())
`

**Comment by MehdiTouns31** [comment]
I failed to run this piece of code on my Hassio server because in the terminal the "python3" command is not recognized ... 

Is there another way to test ? 

We miss prayer times at home.
 Thanks for your help.

**Comment by MehdiTouns31** [comment]
Finally I was able to run the test.py script on a Linux machine. Here is the result returned:
"
[X@X Téléchargements]$ python3 test.py
Collecting mawaqit_times_calculator
  Using cached https://files.pythonhosted.org/packages/e6/6c/1324a1cad5e9be125c8e8851e2187c5241348deb1a70ce2733271f1c5681/mawaqit_times_calculator-0.0.4.tar.gz
Installing collected packages: mawaqit-times-calculator
  Running setup.py install for mawaqit-times-calculator ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-0nxwcz0b/mawaqit-times-calculator/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-jn7so5p6-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib
    creating build/lib/mawaqit_times_calculator
    copying mawaqit_times_calculator/pray_times_calculator.py -> build/lib/mawaqit_times_calculator
    copying mawaqit_times_calculator/__init__.py -> build/lib/mawaqit_times_calculator
    running install_lib
    creating /usr/local/lib/python3.6
    error: could not create '/usr/local/lib/python3.6': Permission denied
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-0nxwcz0b/mawaqit-times-calculator/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-jn7so5p6-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-0nxwcz0b/mawaqit-times-calculator/
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    from mawaqit_times_calculator import MawaqitTimesCalculator
ModuleNotFoundError: No module named 'mawaqit_times_calculator'
"

I tried running it with super user rights as well:
"
[root@X Téléchargements]# python3 test.py
WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Collecting mawaqit_times_calculator
  Downloading https://files.pythonhosted.org/packages/e6/6c/1324a1cad5e9be125c8e8851e2187c5241348deb1a70ce2733271f1c5681/mawaqit_times_calculator-0.0.4.tar.gz
Installing collected packages: mawaqit-times-calculator
  Running setup.py install for mawaqit-times-calculator ... done
Successfully installed mawaqit-times-calculator-0.0.4
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    from mawaqit_times_calculator import MawaqitTimesCalculator
ModuleNotFoundError: No module named 'mawaqit_times_calculator'
"

Do you have an idea ?

**Comment by yac69** [comment]
Salam Mehdi.
sorry for the late reply. I think I get the issue.
what is the folder name of mawaqit in the custom_components directory in hassio

rename it mawaqit_prayer_times and check if it works please.

**Comment by MehdiTouns31** [comment]
Wa3leykoum Salam,

After checking, the name of the mawaqit folder in custom_components of my server is "mawaqit_prayer_times".

I tried something else:
I uninstalled the mawaqit integration and deleted the mawaqit_prayer_times folder from my server.
I updated to the latest version of Home Assistant OS and Core on my server. Then I redownloaded mawaqit and installed the integration. I entered my configuration and chose my mosque and there it worked. I found my prayer sensors. I don't understand why it didn't work before, but the main thing is that now it works. Barrakallahufik for your help. See you soon.
